vcd/resource_vcd_cloned_vapp.go

```diff
@@ -0,0 +1,247 @@
+		}
+		params := &types.InstantiateVAppTemplateParams{
+			Name:        vappName,
+			Deploy:      true,
+			PowerOn:     powerOn,
+			Description: vappDescription,
+			Source: &types.Reference{
+				HREF: sourceTemplate.VAppTemplate.HREF,
+				ID:   sourceTemplate.VAppTemplate.ID,
+				Type: sourceTemplate.VAppTemplate.Type,
+				Name: sourceTemplate.VAppTemplate.Name,
+			},
+			IsSourceDelete:   deleteSource,
+			AllEULAsAccepted: true,
+		}
+		vapp, err = vdc.CreateVappFromTemplate(params)
+		if err != nil {
+			return diag.Errorf("error creating vApp %s from template: %s", vappName, err)
+		}
+	}
+
+	d.SetId(vapp.VApp.ID)
+
+	vappStatus, err := vapp.GetStatus()
+	if err != nil {
+		return diag.Errorf("error retrieving vApp '%s' status: %s", vappName, err)
+	}
+	// If the cloned vApp is suspended and power ON was requested, we adjust the status accordingly
+	if vappStatus == "SUSPENDED" && powerOn {
+		err = vapp.DiscardSuspendedState()
+		if err != nil {
+			return diag.Errorf("error requesting suspended state change on vApp '%s': %s", vappName, err)
+		}
+		task, err := vapp.PowerOn()
+		if err != nil {
+			return diag.Errorf("error requesting power change on vApp '%s': %s", vappName, err)
+		}
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return diag.Errorf("error while powering on vApp '%s': %s", vappName, err)
+		}
+	}
+
+	return resourceVcdClonedVAppRead(ctx, d, meta)
+}
+
+func resourceVcdClonedVAppRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return diag.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+	identifier := d.Id()
+
+	if identifier == "" {
+		identifier = d.Get("name").(string)
+	}
+	if identifier == "" {
+		return diag.Errorf("[vapp read] no identifier provided")
+	}
+	vapp, err := vdc.GetVAppByNameOrId(identifier, false)
+	if err != nil {
+		log.Printf("[DEBUG] Unable to find vApp. Removing from tfstate")
+		d.SetId("")
+		return nil
+	}
+
+	statusText, err := vapp.GetStatus()
+	if err != nil {
+		statusText = vAppUnknownStatus
+	}
+	dSet(d, "status", vapp.VApp.Status)
+	dSet(d, "status_text", statusText)
+	dSet(d, "href", vapp.VApp.HREF)
+	dSet(d, "description", vapp.VApp.Description)
+
+	var vmList []string
+	if vapp.VApp.Children != nil {
+		for _, vm := range vapp.VApp.Children.VM {
+			vmList = append(vmList, vm.Name)
+		}
+		sort.Strings(vmList)
+	}
+
+	err = d.Set("vm_list", vmList)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(vapp.VApp.ID)
+	return nil
+}
+
+func resourceVcdClonedVAppDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceVcdVAppDelete(ctx, d, meta)
+}
+
+func resourceVcdClonedVAppUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return diag.Errorf("this resource doesn't support updates")
+}
```

Comment:
NIT, but you could set `ForceNew` on all non computed attributes and skip specifying `update` function at all.

Reply:
I was undecided between these two strategies. I reverted to using Forcenew. Thanks